internal/functions/efgs/keys-operations_test.go

```diff
@@ -0,0 +1,233 @@
+		wantMatchResult bool
+		wantChunksCount int
+	}{
+		{
+			name: "basic",
+			args: args{
+				keys: []ExpKey{
+					{Key: "a", IntervalNumber: 10, IntervalCount: 0, TransmissionRisk: 0},
+					{Key: "b", IntervalNumber: 0, IntervalCount: 0, TransmissionRisk: 0},
+					{Key: "c", IntervalNumber: 10, IntervalCount: 0, TransmissionRisk: 0},
+					{Key: "d", IntervalNumber: 10, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "e", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "f", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "g", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "h", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+				},
+				batchSize:      50,
+				maxOverlapping: 2,
+			},
+			wantChunks: [][]ExpKey{
+				{
+					{Key: "b", IntervalNumber: 0, IntervalCount: 0, TransmissionRisk: 0},
+					{Key: "g", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+				},
+				{
+					{Key: "e", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "f", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "d", IntervalNumber: 10, IntervalCount: 2, TransmissionRisk: 0},
+				},
+				{
+					{Key: "h", IntervalNumber: 0, IntervalCount: 2, TransmissionRisk: 0},
+					{Key: "c", IntervalNumber: 10, IntervalCount: 0, TransmissionRisk: 0},
+					{Key: "a", IntervalNumber: 10, IntervalCount: 0, TransmissionRisk: 0},
+				},
+			},
+			wantMatchResult: true,
+			wantChunksCount: 3,
+		},
+		{
+			name: "efgsDownloadedKeys",
+			args: args{
+				keys:           loadTestingKeys("aligned"),
+				batchSize:      50,
+				maxOverlapping: 15,
+			},
+			wantChunks:      nil,
+			wantMatchResult: false,
+			wantChunksCount: 101,
+		},
+		{
+			name: "efgsDownloadedKeysUnaligned",
+			args: args{
+				keys:           loadTestingKeys("unaligned"),
+				batchSize:      50,
+				maxOverlapping: 15,
+			},
+			wantChunks:      nil,
+			wantMatchResult: false,
+			wantChunksCount: 132,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotChunks := splitKeys(tt.args.keys, tt.args.batchSize, tt.args.maxOverlapping)
+
+			if len(gotChunks) != tt.wantChunksCount {
+				t.Errorf("splitKeys() check count error; got %v, want %v", len(gotChunks), tt.wantChunksCount)
+				return
+			}
+
+			for _, chunk := range gotChunks {
+				if err := checkKeyServerBatch(chunk, tt.args.batchSize, tt.args.maxOverlapping); err != nil {
+					t.Errorf("splitKeys() check error = %+v", err)
+					return
+				}
+			}
+
+			t.Logf("Chunks: %v", len(gotChunks))
+
+			if tt.wantMatchResult && !reflect.DeepEqual(gotChunks, tt.wantChunks) {
+				t.Errorf("splitKeys() gotChunks = %+v, want %+v", gotChunks, tt.wantChunks)
+			}
+		})
+	}
+}
+
+func loadTestingKeys(suffix string) (efgsDownloadedKeys []ExpKey) {
+	bytes, err := ioutil.ReadFile(fmt.Sprintf("../../../testdata/efgs-downloaded-keys-%v.json", suffix))
```

Comment:
I'd move the `testdata` folder to `test/data` as per https://github.com/golang-standards/project-layout#test
I have some `*.json` req/resp samples in `example` folder but I'd move them to the new `test/data` too :)